backend/Controllers/MovieController.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using CinemaApp.Models;
+
+namespace CinemaApp.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class MovieController : ControllerBase
+    {
+        private readonly MovieContext _context;
+
+        public MovieController(MovieContext context)
+        {
+            _context = context;
+        }
+
+        // GET: api/Movie
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<Movie>>> Getmovies()
+        {
+            return await _context.movies.ToListAsync();
+        }
+
+        // GET: api/Movie/5
+        [HttpGet("{id}")]
+        public async Task<ActionResult<Movie>> GetMovie(int id)
+        {
+            var movie = await _context.movies.FindAsync(id);
+
+            if (movie == null)
+            {
+                return NotFound();
+            }
+
+            return movie;
+        }
+
+        
+        [HttpPut("{id}")]
+        public async Task<IActionResult> PutMovie(int id, Movie movie)
+        {
+            if (id != movie.ID)
+            {
+                return BadRequest();
+            }
+
+            _context.Entry(movie).State = EntityState.Modified;
+
+            try
+            {
+                await _context.SaveChangesAsync();
+            }
+            catch (DbUpdateConcurrencyException)
+            {
+                if (!MovieExists(id))
+                {
+                    return NotFound();
+                }
+                else
+                {
+                    throw;
+                }
+            }
+
+            return NoContent();
+        }
+
+        // POST: api/Movie
+        // To protect from overposting attacks, enable the specific properties you want to bind to, for
+        // more details, see https://go.microsoft.com/fwlink/?linkid=2123754.
+        [HttpPost]
+        public async Task<ActionResult<Movie>> PostMovie(Movie movie)
+        {
+            _context.movies.Add(movie);
```

Comment:
See my previous comment. I can send you here a model with a negative ID value, or with an existing ID value - you are directly saving it.

Reply:
I already talked with my colleague who made the Movie class about this problem. He will solve itz